File: examples/apps/react-native/src/features/auth/domain/repositories/AuthRepository.ts
import { AuthSession, User } from '../entities/User';

export interface AuthRepository {
  login(email: string, password: string): Promise<AuthSession>;
  register(name: string, email: string, password: string): Promise<AuthSession>;
  updateProfile(name: string, email: string, phone: string): Promise<User>;
}

File: examples/apps/flutter/lib/features/auth/domain/repositories/auth_repository.dart
import 'package:app_prueba_mobilelab/features/auth/domain/entities/app_user.dart';

abstract interface class AuthRepository {
  Future<AuthSession> login({required String email, required String password});

  Future<AuthSession> register({
    required String name,
    required String email,
    required String password,
  });

  Future<AppUser> getProfile();

  Future<AppUser> updateProfile({
    required String name,
    required String email,
    required String phone,
  });
}
